lib/mysql2.rb

```diff
@@ -3,6 +3,26 @@
 require 'bigdecimal'
 require 'rational' unless RUBY_VERSION >= '1.9.2'
 
+# Load libmysql.dll before requiring mysql2/mysql2.so
+# This gives a chance to be flexible about the load path
+# Or to bomb out with a clear error message instead of a linker crash
+if RUBY_PLATFORM =~ /mswin|mingw/
+  dll_search = [
+    File.expand_path('../vendor/libmysql.dll', File.dirname(__FILE__)).gsub('/', '\\'),
+    File.expand_path('../libmysql.dll', File.dirname(__FILE__)).gsub('/', '\\'),
```

Comment:
Update: the fat binary mechanism is for multiple Ruby versions for the same platform, which means they would share the same `vendor/libmysql.dll` binary. I should still remove the second search location, though.